zcash_client_backend/Cargo.toml

```diff
@@ -68,6 +68,11 @@ group.workspace = true
 orchard = { workspace = true, optional = true }
 sapling.workspace = true
 
+# - Sync engine
+anyhow = { version = "1", optional = true }
```

Comment:
I hope you'll avoid `anyhow` errors. Use an enum type with members for each error condition. That way the app can reason about what went wrong and make a decision about how to recover, or what to communicate with the user.

Reply:
This PR is a draft for a reason: I threw it together having extracted it from an _application_, where `anyhow` is perfectly acceptable to use. Part of the work to do to make this ready for review will indeed be to have error types that are useful for this in a library context.